### ISSUE TITLE
[🐞] CORS error for Google Tag Manager Service Worker

### ISSUE THREAD

**Issue by brandonfredericksen** [QwikDev/partytown#635]
### Describe the bug

I'm getting the following CORS error when using Partytown and Google Tag Manager.

`Access to XMLHttpRequest at 'https://www.googletagmanager.com/static/service_worker/4al0/sw_iframe.html?origin=https%3A%2F%2Fwww.url.com' from origin 'https://www.url.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

![image](https://github.com/user-attachments/assets/9fe236c6-8a43-4aac-b43c-4e3958908255)

Error happens in production as well.

Current implementation:

```
<Partytown
  debug={false}
  forward={['dataLayer.push']}
  loadScriptsOnMainThread={[
    'https://googleads.g.doubleclick.net/pagead/viewthroughconversion',
  ]}
/>
```

I'm using Remix. I've tried adding the url to the `loadScriptsOnMainThread` without success:

```
<Partytown
  debug={false}
  forward={['dataLayer.push']}
  loadScriptsOnMainThread={[
    'https://googleads.g.doubleclick.net/pagead/viewthroughconversion',
    'https://www.googletagmanager.com/static/service_worker', // doesn't work
  ]}
/>
```

### Reproduction

private

### Steps to reproduce

Use Partytown with Google Tag Manager

### Browser Info

Chrome

**Comment by andreja-migles-sm** [comment]
Having the same issue.

**Comment by jrood** [comment]
Indeed, when gtm.js tries to create this iframe, Partytown seems to be proxying this behavior and doing something else (perhaps requesting the iframe src) instead. Since, this src matches `loadScriptsOnMainThread`, I would expect Partytown to allow gtm.js to create the iframe in the dom normally, but it does not.

Even if Google were to return cors headers for `sw_iframe.html` or if this is mitigated with a reverse proxy, I anticipate another issue. Google's code within this iframe seems to be attempting to register a service worker itself, and I'm not sure if Partytown will allow this type of service worker incpetion.

**Comment by jrood** [comment]
It's also not clear if `sw_iframe.html` is even necessary. Tags still seem to load without it.

**Comment by tcp2** [comment]
@jrood 
The `page_view` data collected from Google Analytics, originating from Facebook, has decreased by half.

**Comment by jrood** [comment]
Ok yes, `sw_iframe` seems necessary for running sandboxed code. If Partytown could honor `loadScriptsOnMainThread` for iFrames (as it does currently for scripts), I believe this would resolve the issue.

**Comment by hnsxxscyx** [comment]
@jrood @brandonfredericksen @tcp2 
Hi guys, have you solved this issue? Any advice for this iframe?

**Comment by HumboldtK** [comment]
Same issue, I see websites and articles recommending Partytown to load GTM but I get this error.

**Comment by muhsin-life** [comment]
try to implement reverse proxy

**Comment by beumerr** [comment]
Same problem here. I can confirm the error still occurs while using a reverse proxy connected to https://www.googletagmanager.com/gtm.js?id= or https://www.googletagmanager.com/gtag/js?id=

**Comment by muhsin-life** [comment]
please produce a minimum reproducible example

**Comment by devcaeg** [comment]
Same problem here

**Comment by QuantumQuin** [comment]
I had to remove partytown due to this issue, such a shame as it really boosted performance

**Comment by selastingeorge** [comment]
Is this resolved?, I'm having same CORS issue.

**Comment by prabapro** [comment]
Same here. Started seeing CORS error ever since Google started using service workers for server side tagging.

[GTM release notes](https://support.google.com/tagmanager/answer/4620708#:~:text=in%2Dpage%20code.-,March%204%2C%202025,-Google%20tag%20now)

> March 4, 2025
Google tag now uses service workers, when available, to send data to server-side Tag Manager, improving performance and measurement reliability.

**Comment by Conrado2011** [comment]
any?

**Comment by paraspant09** [comment]
I am stuck with same issue can somebody help?

**Comment by thomas1001-droid** [comment]
Any fix or way to fix this issue ?

**Comment by petrch87** [comment]
I am afraid that this package is not being maintained anymore, not actively anyway. Look at insights, it has basically no activity.

**Comment by gioboa** [comment]
Likely fixed by #753: resources the worker can't read due to missing CORS headers now fall back to the main thread instead of failing.
So I'm closing this issue for now, feel free to re-open it if it's still an issue for you.
